File: 14. Longest Common Prefix/solution.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if "" in strs or len(strs) == 0:
            return ""
        else:
            ans = ""
            for i in range(len(strs[0])):
                flag = 1
                for string in strs:
                    if len(string) <= i:
                        flag = 0
                        break
                    else:
                        if strs[0][i] == string[i]:
                            continue
                        else:
                            flag = 0
                            break
                if flag:
                    ans += strs[0][i]
                else:
                    break
            return ans

File: 14. Longest Common Prefix/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        
        int i = 0, flag = 1;
        while(flag){
           for(int j = 0; j < strs.size(); j++){
               if(i >= strs[j].size() || strs[j][i] != strs[0][i]){
                   flag = 0;
                   break;}}
           i++;}
        string res = strs[0].substr(0, i-1);
        return res;
    }
};
